### PR TITLE
add size_on_disk() to file_storage and torrent_info

### PR DESCRIPTION
**Pull request by arvidn**
reporting the total size of non-pad files

### CODE REVIEW

File: src/file_storage.cpp [hunk 1]
```diff
@@ -740,6 +740,8 @@ namespace aux {
 		}
 
 		m_total_size += e.size;
+		m_size_on_disk += e.size;
```
Comment (master255) [inline src/file_storage.cpp:743]: Where is the logic?

File: src/file_storage.cpp [hunk 2]
```diff
@@ -1223,13 +1227,16 @@ namespace {
 			TORRENT_ASSERT(off < max_file_offset - static_cast<std::int64_t>(file.size));
 			file.offset = static_cast<std::uint64_t>(off);
 			off += file.size;
+			on_disk += file.size;
```
Comment (master255) [inline src/file_storage.cpp:1230]: Where is the logic?

File: test/test_file_storage.cpp [hunk 3]
```diff
@@ -52,6 +52,7 @@ void setup_test_storage(file_storage& st)
 	TEST_EQUAL(st.file_offset(file_index_t{3}), 60000);
 
 	TEST_EQUAL(st.total_size(), 100000);
+	TEST_EQUAL(st.size_on_disk(), 100000);
```
Comment (master255) [inline test/test_file_storage.cpp:55]: For what purpose make another variable, if it will also be equal to total_size

Comment (arvidn) [reply test/test_file_storage.cpp:55]: As I've already explained, the total size and size_on_disk are the same for torrents with no pad files.

```
total_size == size_on_disk + pad_file_size
```

Comment (master255) [reply test/test_file_storage.cpp:55]: @arvidn Maybe I'm not seeing something. Where do you filter .pad files?

Comment (master255) [reply test/test_file_storage.cpp:55]: @arvidn Okay. I found where the pad files are filtered. But you have to test my .torrent file. The information in it doesn't match. Maybe there are some other problems.

**Pull request merged**